update.ps1

```diff
@@ -160,208 +107,239 @@ function Resolve-ZivverError {
-                if ($key -eq "urn:ietf:params:scim:schemas:zivver:0.1:User") {
-                    $array1 = $referenceValue -join ","
-                    $array2 = $differenceValue -join ","
-                    $differences += "Array1: $array1"
-                    $differences += "Array2: $array2"
-                }
-            }
-        }
-        elseif ($referenceValue -ne $differenceValue) {
-            $differences += $key
-        }
-    }
-
-    Write-Output $differences
-}
-
-function Compare-Array {
-    param (
-        [object]
-        $ReferenceObject,
-
-        [object]
-        $DifferenceObject
-    )
-
-    if ($ReferenceObject.Length -ne $DifferenceObject.Length) {
-        return $false
+try {
+    #region Verify account reference
+    $actionMessage = "verifying account reference"
+    if ([string]::IsNullOrEmpty($($actionContext.References.Account))) {
+        throw "The account reference could not be found"
     }
+    #endregion Verify account reference
 
-    $sortedArr1 = $ReferenceObject | Sort-Object
-    $sortedArr2 = $DifferenceObject | Sort-Object
+    #region Create authorization headers
+    $actionMessage = "creating authorization headers"
 
-    for ($i = 0; $i -lt $sortedArr1.Length; $i++) {
-        if ($sortedArr1[$i] -ne $sortedArr2[$i]) {
-            return $false
-        }
-    }
+    $headers = [System.Collections.Generic.Dictionary[[String], [String]]]::new()
+    $headers.Add("Authorization", "Bearer $($actionContext.Configuration.Token)")
+    Write-Verbose "Created authorization headers"
+    #endregion Create authorization headers
 
-    return $true
-}
-#endregion
+    #region Get Zivver account
+    $actionMessage = "querying Zivver account"
 
-# Begin
-try {
-    Write-Verbose "Verify if [$aRef] has a value"
-    if ([string]::IsNullOrEmpty($($aRef))) {
-        throw 'Mandatory attribute [aRef] is empty.'
+    $getZivverSplatParams = @{
+        Headers  = $headers
+        Endpoint = "Users/$($actionContext.References.Account)"
+        Method   = 'GET'
     }
-
-    Write-Verbose 'Creating authorization header'
-    $headers = [System.Collections.Generic.Dictionary[[String], [String]]]::new()
-    $headers.Add("Authorization", "Bearer $($config.Token)")
-    $splatParams = @{
-        Headers = $headers
-    }
-
     try {
-        Write-Verbose "Verifying if a Zivver account for [$($p.DisplayName)] exists"
-        $splatParams['Endpoint'] = "Users/$aRef"
-        $splatParams['Method'] = 'GET'
-        $responseUser = Invoke-ZivverRestMethod @splatParams
+        $correlatedAccount = Invoke-ZivverRestMethod @getZivverSplatParams
     }
     catch {
         # A '400'bad request is returned if the entity cannot be found
         if ($_.Exception.Response.StatusCode -eq 400) {
-            $responseUser = $null
+            $correlatedAccount = $null
         }
         else {
             throw
         }
     }
+    $outputPreviousData = $correlatedAccount.PsObject.Copy()
```

Comment:
In principe zou deze niet nodig hoeven zijn. `PSObject.Copy()`.  We passen de outputPreviousData nergens meer aan zo te zien. 

En, iets om rekening mee te houden, `.Copy()` kan niet omgaan met hele diepe/gelaagde structuren. Dan zou je er eerst een JSON van kunnen maken.

Reply:
Zoals besproken. We verreiken $correlatedAccount om een volledige body terug te sturen naar Zivver. Daarom de Copy.